brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -222,10 +242,23 @@ public static void set(TraceContext context, String name, String value) {
     final Propagation.Factory delegate;
     final String[] fieldNames;
     final String[] keyNames;
+    final int[] keyToField;
     final ExtraFactory extraFactory;
 
     Factory(Propagation.Factory delegate, String[] fieldNames, String[] keyNames) {
+      this(delegate, fieldNames, keyNames, keyToField(keyNames));
+    }
+
+    static int[] keyToField(String[] keyNames) {
```

Comment:
somewhere there should be a javadoc saying:

"we have a key to field mapping as there may be multiple propagation keys that reference the same field. For example, baggage-userid and baggage_userid"